### ISSUE TITLE
simple patch for scratchpad position

### ISSUE THREAD

**Issue by tramhao** [leftwm/leftwm#483]
Probably we need more control of the size and position of scratchpad, and here is my patch for a scratch pad on right bottom which is my personal preference. I'm not sure if there is any plan on this issue so I posted it here. If you guys have any suggestions, please kindly let me know. Since this is quite personal, I'll not submit a PR until it's considered more seriously.

```
diff --git a/src/handlers/window_handler.rs b/src/handlers/window_handler.rs
index 03ccbe2..4903302 100644
--- a/src/handlers/window_handler.rs
+++ b/src/handlers/window_handler.rs
@@ -85,7 +85,7 @@ fn setup_window(
 
         if is_scratchpad {
             window.set_floating(true);
-            let new_float_exact = ws.center_halfed();
+            let new_float_exact = ws.right_bottom();
             window.normal = ws.xyhw;
             window.set_floating_exact(new_float_exact);
         }
diff --git a/src/models/workspace.rs b/src/models/workspace.rs
index 3c4dc40..69e1351 100644
--- a/src/models/workspace.rs
+++ b/src/models/workspace.rs
@@ -289,6 +289,10 @@ impl Workspace {
         self.xyhw_avoided.center_halfed()
     }
 
+    pub fn right_bottom(&self) -> Xyhw {
+        self.xyhw_avoided.right_bottom_corner()
+    }
+
     pub fn update_avoided_areas(&mut self) {
         let mut xyhw = self.xyhw;
         for a in &self.avoid {
diff --git a/src/models/xyhw.rs b/src/models/xyhw.rs
index 4f4f9ce..e612da2 100644
--- a/src/models/xyhw.rs
+++ b/src/models/xyhw.rs
@@ -268,6 +268,17 @@ impl Xyhw {
         .into()
     }
 
+    pub fn right_bottom_corner(&self) -> Xyhw {
+        XyhwBuilder {
+            x: self.x + self.w - (self.w / 2 - self.w / 20),
+            y: self.y + (self.h / 5), //- (self.h / 4),
+            h: (self.h - self.h / 5),
+            w: (self.w / 2 - self.w / 20),
+            ..XyhwBuilder::default()
+        }
+        .into()
+    }
+
     pub fn center_relative(&mut self, outer: Self, border: i32, change: Option<XyhwChange>) {
         // If this is a Splash screen, for some reason w/h are sometimes not sized properly. We
         // therefore want to use the minimum or maximum sizing instead in order to center the
```

**Comment by VuiMuich** [comment]
Welcome Adventurer!

Thanks for this patch.
We recently discussed about additional control for scratchpads; will have to look up the issue ids later.
But basically the ideas where configurable size and position in pixels, making use of a recently introduced hiding position (would need to be extended for hiding at top and bottom) so in combination with picom-jonaburg scratchpads could mimic drop down terminals.

If you could extend your patch so it gets configurable, and maybe add a bunch other dynamic positions a PR would surely be very welcome!

**Comment by tramhao** [comment]
merged #493

**Comment by VuiMuich** [comment]
Thanks for your contribution!
